tests/graph_components/validators/test_finetuning_validator.py

```diff
@@ -0,0 +1,434 @@
+from rasa.shared.exceptions import InvalidConfigException
+from rasa.shared.nlu.constants import ACTION_NAME, INTENT
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.training_data.training_data import TrainingData
+
+
+@pytest.fixture
+def default_resource() -> Resource:
+    return Resource("FineTuningValidator")
+
+
+ValidationMethodType = Callable[[TrainingDataImporter, Dict[Text, Any]], None]
+
+
+@pytest.fixture
+def get_finetuning_validator(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    default_resource: Resource,
+) -> Callable[[bool, bool], FineTuningValidator]:
+    def inner(finetuning: bool, load: bool,) -> FineTuningValidator:
+        if load:
+            constructor = FineTuningValidator.load
+        else:
+            constructor = FineTuningValidator.create
+        if finetuning:
+            default_execution_context.is_finetuning = finetuning
+        return constructor(
+            config=FineTuningValidator.get_default_config(),
+            execution_context=default_execution_context,
+            model_storage=default_model_storage,
+            resource=default_resource,
+        )
+
+    return inner
+
+
+@pytest.fixture
+def get_validation_method(
+    get_finetuning_validator: Callable[[bool, bool], FineTuningValidator],
+    nlu: bool,
+    core: bool,
+) -> Callable[[bool, bool, bool, bool], ValidationMethodType]:
+    def inner(
+        finetuning: bool, load: bool, nlu: bool, core: bool
+    ) -> ValidationMethodType:
+        validator = get_finetuning_validator(finetuning=finetuning, load=load)
+        if core and nlu:
+            method = "validate"
+        elif core:
+            method = "validate_core_only"
+        elif nlu:
+            method = "validate_nlu_only"
+        else:
+            method = "validate"
+        func = getattr(validator, method)
+        if not core and not nlu:
+            func = functools.partial(func, nlu=False, core=False)
+        return func
+
+    return inner
+
+
+def _project_files(
+    project: Text,
+    config_file: Text = DEFAULT_CONFIG_PATH,
+    domain: Text = DEFAULT_DOMAIN_PATH,
+    training_files: Text = DEFAULT_DATA_PATH,
+) -> TrainingDataImporter:
+    paths = {
+        "config_file": config_file,
+        "domain_path": domain,
+        "training_data_paths": training_files,
+    }
+    paths = {
+        k: v if v is None or Path(v).is_absolute() else os.path.join(project, v)
+        for k, v in paths.items()
+    }
+    paths["training_data_paths"] = [paths["training_data_paths"]]
+
+    return RasaFileImporter(**paths)
+
+
+@pytest.mark.parametrize("nlu, core", [(True, False), (False, True), (True, True)])
+def test_validate_after_changing_response_text_in_domain(
+    get_validation_method: Callable[..., ValidationMethodType],
+    project: Text,
+    nlu: bool,
+    core: bool,
+):
+    # NOTE: test_model/test_validate_changed_response_text
```

Comment:
What does this mean?

Reply:
This is just for us / for the review to see where a check came from